packages/solid-query/src/useBaseQuery.ts

```diff
@@ -159,7 +156,7 @@ export function useBaseQuery<
         const query = observer().getCurrentQuery()
         const unwrappedResult = hydratableObserverResult(query, result)
 
-        if (unwrappedResult.isError) {
+        if (result.data !== undefined && unwrappedResult.isError) {
```

Comment:
this only skips it for the server subscriber. I am not sure if there is a better heuristic here instead of just data is undefined